app/services/participants/change_schedule/ecf.rb

```diff
@@ -3,8 +3,13 @@
 module Participants
   module ChangeSchedule
     class ECF < Base
-      include Participants::ECF
-      include ValidateAndChangeSchedule
+      delegate :school_cohort, to: :user_profile, allow_nil: true
+
+    private
+
+      def matches_lead_provider?
+        cpd_lead_provider == school_cohort&.lead_provider&.cpd_lead_provider
```

Comment:
Suggestion: it think this reads much better
```ruby
return false unless school_cohort&.lead_provider

cpd_lead_provider == school_cohort.lead_provider.cpd_lead_provider
```

Reply:
I've mostly left these as is for now